DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/Cluster.h

```diff
@@ -0,0 +1,102 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef ALICEO2_EMCAL_CLUSTER_H_
+#define ALICEO2_EMCAL_CLUSTER_H_
+
+#include <array>
+#include <exception>
+#include <iosfwd>
+#include <string>
+#include <vector>
+#include "CommonDataFormat/TimeStamp.h"
+#include "CommonDataFormat/RangeReference.h"
+#include "MathUtils/Cartesian3D.h"
+
+namespace o2
+{
+
+namespace emcal
+{
+
+/// \class Cluster
+/// \brief EMCAL Cluster
+///
+class Cluster : public o2::dataformats::TimeStamp<Float16_t>, public Point3D<Float16_t>
+{
```

Comment:
One point more: if you stay with Float16_t, I think you should add 
``#pragma link C++ class o2::dataformats::TimeStamp < Float16_t > + ;``
to ``DataFormats/common/src/CommonDataFormatLinkDef.h``

Reply:
Added, under the assumption we stay with Float16_t for the cluster time.